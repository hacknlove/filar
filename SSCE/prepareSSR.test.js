const { DOMParser } = require("linkedom");
const parser = new DOMParser();

const { prepareSSR } = require("./prepareSSR");

describe("prepareSSR", () => {
  it("adds a script tag with the context", () => {
    const node = parser.parseFromString("<SSR>{{Title}}</SSR>").firstChild;

    const context = {
      Title: "old title",
    };

    prepareSSR(node, context);

    expect(node.toString()).toMatchSnapshot();
  });

  it("skips the script tag if it already exists", () => {
    const node = parser.parseFromString(
      "<SSR><script type='application/json' class='ssr'></script>{{title}}</SSR>"
    ).firstChild;

    const context = {
      title: "old title",
    };

    prepareSSR(node, context);

    expect(node.toString()).toMatchSnapshot();
  });
});
