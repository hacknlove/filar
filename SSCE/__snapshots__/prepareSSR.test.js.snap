// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepareSSR adds a script tag with the context 1`] = `"<SSR><script class="ssr" type="application/json">{"Title":"old title"}</script>{{Title}}</SSR>"`;

exports[`prepareSSR skips the script tag if it already exists 1`] = `"<SSR><script type="application/json" class="ssr" />{{title}}</SSR>"`;
